bluebutton-data-model-rif/src/main/java/gov/hhs/cms/bluebutton/data/model/rif/RifFilesEvent.java

```diff
@@ -1,31 +1,33 @@
 package gov.hhs.cms.bluebutton.data.model.rif;
 
 import java.time.Instant;
-import java.time.format.DateTimeFormatter;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 
+
+
 /**
  * Models a RIF file event, in which a new set of RIF files have been made
  * available for processing.
  */
 public final class RifFilesEvent {
-	private final Instant timestamp;
+	// private final Instant timestamp;
+	private final DataSetManifest dataSetManifest;
 	private final List<RifFileEvent> fileEvents;
 
 	/**
 	 * Constructs a new {@link RifFilesEvent} instance.
 	 * 
-	 * @param timestamp
-	 *            the value to use for {@link #getTimestamp()}
+	 * @param dataSetManifest
+	 *            the value to use for {@link #getDataSetManifest()}
 	 * @param files
 	 *            the value to use for {@link #getFileEvents()}
 	 */
-	public RifFilesEvent(Instant timestamp, List<RifFile> files) {
-		if (timestamp == null)
+	public RifFilesEvent(DataSetManifest dataSetManifest, List<RifFile> files) {
+		if (dataSetManifest.getTimestamp() == null)
```

Comment:
This check here is a bit odd. Instead, I'd check to see if the `DataSetManifest` itself is `null`. If you also need to ensure that its timestamp isn't ever null, that check should go in its constructor.